src/tests/Loader/classloader/ExtendedLayout/CStruct.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace ExtendedLayoutTests;
+
+public static class CStructTests
+{
+    [Fact]
+    public static void BlittablePrimitiveFieldsLayout()
+    {
+        var c = default(CStructBlittablePrimitiveFields);
+
+        Assert.Equal(12, Unsafe.SizeOf<CStructBlittablePrimitiveFields>());
+        Assert.Equal(0, Unsafe.ByteOffset(ref Unsafe.As<CStructBlittablePrimitiveFields, byte>(ref c), ref Unsafe.As<int, byte>(ref c.a)));
+        Assert.Equal(4, Unsafe.ByteOffset(ref Unsafe.As<CStructBlittablePrimitiveFields, byte>(ref c), ref Unsafe.As<float, byte>(ref c.b)));
+        Assert.Equal(8, Unsafe.ByteOffset(ref Unsafe.As<CStructBlittablePrimitiveFields, byte>(ref c), ref c.c));
+    }
+
+    [Fact]
+    public static void NonBlittableUnmanagedPrimitiveFields_TreatedAsBlittable()
+    {
+        Assert.Equal(Unsafe.SizeOf<CStructNonBlittablePrimitiveFields>(), Marshal.SizeOf<CStructNonBlittablePrimitiveFields>());
```

Comment:
This type should have a stable size we can hardcode to, basically it is expected to be X. Since we're following the C rules it should be explicitly stated. I don't know of any guarantees that say `Marshal.SizeOf<>` is always the C rules. It is implied all over, but I'd prefer explicit sizes stated when possible.

Reply:
This test is verifying that we consider the type as blittable and don’t generate a “native” struct type with the default marshalling rules for bool and char.

I’ll add asserts for the size and offsets against known values.